src/components/Filters/Filters.tsx

```diff
@@ -5,6 +5,60 @@ import 'material-icons/iconfont/material-icons.css'
 import { FilterButton } from '@/components/FilterButton/FilterButton'
 
 const Filters = () => {
+  const [selectedTechnology, setSelectedTechnology] = useState<string[]>([])
+  const [selectedSeniority, setSelectedSeniority] = useState<string[]>([])
+  const [selectedAvailability, setSelectedAvailability] = useState<string[]>([])
+  const [selectedLocation, setSelectedLocation] = useState<string[]>([])
+
+  const handleSelect = (option: string, buttonType: string) => {
+    let updatedOptions: string[]
+
+    switch (buttonType) {
+      case 'Technology':
+        updatedOptions = selectedTechnology.includes(option)
+          ? selectedTechnology.filter(
+              (selectedOption) => selectedOption !== option,
+            )
+          : [...selectedTechnology, option]
+        setSelectedTechnology(updatedOptions)
+        console.log(selectedTechnology)
+        break
+
+      case 'Seniority':
+        updatedOptions = selectedSeniority.includes(option)
+          ? selectedSeniority.filter(
+              (selectedOption) => selectedOption !== option,
+            )
+          : [...selectedSeniority, option]
+        setSelectedSeniority(updatedOptions)
+        console.log(selectedSeniority)
+        break
+
+      case 'Availability':
```

Comment:
**nitpick:** You could keep these values as enum, it's quite important, and we will avoid string duplication / maintanance:

```ts
export enum FilterNames {
Availability = "Availability",
...
}